mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -54,6 +54,9 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
     const [unifiedDiffLines, setUnifiedDiffLines] = useState<UnifiedDiffLine[] | undefined>(undefined)
     const originalCodeBeforeDiff = useRef<string | undefined>(undefined)
 
+    const [isApplyingCode, setIsApplyingCode] = useState<boolean>(false);
+    const [codeWasAccepted, setCodeWasAccepted] = useState<boolean>(false);
```

Comment:
Let's combine these state variable into something more like a state machine. We could call the variable: `codeReviewStatus`. We're either in:
1.  `chat preview` state where the user has not yet pressed the `apply` button
2. `code cell preview` state where the user is seeing the code diffs and deciding how they want to respond
3. `applied` state where the user has applied the code to the code cell